File: ru.kmorozov.gbd.core.logger/src/main/java/ru/kmorozov/gbd/logger/consumers/IEventConsumer.java
package ru.kmorozov.gbd.logger.consumers;

import ru.kmorozov.gbd.logger.events.BaseEvent;
import ru.kmorozov.gbd.logger.listeners.IEventListener;

/**
 * Created by km on 15.12.2015.
 */
public interface IEventConsumer {

    void addListener(IEventListener listener);

    void consumeEvent(BaseEvent event);
}

File: ru.kmorozov.gbd.core.logger/src/main/java/ru/kmorozov/gbd/logger/consumers/IEventConsumer.kt
package ru.kmorozov.gbd.logger.consumers

import ru.kmorozov.gbd.logger.events.LogEvent
import ru.kmorozov.gbd.logger.listeners.IEventListener

/**
 * Created by km on 15.12.2015.
 */
interface IEventConsumer {

    fun addListener(listener: IEventListener)

    fun consumeEvent(event: LogEvent)
}
